packages/service-worker/src/update.ts

```diff
@@ -50,29 +74,54 @@ export class SwUpdate {
 
   constructor(private sw: NgswCommChannel) {
     if (!sw.isEnabled) {
+      this.versionUpdates = NEVER;
       this.available = NEVER;
       this.activated = NEVER;
       this.unrecoverable = NEVER;
       return;
     }
-    this.available = this.sw.eventsOfType<UpdateAvailableEvent>('UPDATE_AVAILABLE');
+    this.versionUpdates = this.sw.eventsOfType<VersionEvent>(
+        ['VERSION_DETECTED', 'VERSION_INSTALLATION_FAILED', 'VERSION_READY']);
+    this.available = this.versionUpdates.pipe(
+        filter((evt: VersionEvent): evt is VersionReadyEvent => evt.type === 'VERSION_READY'),
+        map(evt => ({
+              type: 'UPDATE_AVAILABLE',
+              current: evt.currentVersion,
+              available: evt.latestVersion,
+            })));
     this.activated = this.sw.eventsOfType<UpdateActivatedEvent>('UPDATE_ACTIVATED');
     this.unrecoverable = this.sw.eventsOfType<UnrecoverableStateEvent>('UNRECOVERABLE_STATE');
   }
 
-  checkForUpdate(): Promise<void> {
+  /**
+   * Checks for an update and waits until the new version is ready for activation.
+   *
+   * @returns a promise that
+   * - resolves to `true` if a new version was found and is ready to be activated.
+   * - resolves to `false` if no new version was found
+   * - rejects if any error occurs
+   */
+  checkForUpdate(): Promise<boolean> {
     if (!this.sw.isEnabled) {
       return Promise.reject(new Error(ERR_SW_NOT_SUPPORTED));
     }
-    const statusNonce = this.sw.generateNonce();
-    return this.sw.postMessageWithStatus('CHECK_FOR_UPDATES', {statusNonce}, statusNonce);
+    const nonce = this.sw.generateNonce();
+    return this.sw.postMessageWithOperation('CHECK_FOR_UPDATES', {nonce}, nonce);
   }
 
-  activateUpdate(): Promise<void> {
+  /**
+   * Activates an update that is ready for activation.
```

Comment:
Let's make it clear it only affects the calling client (not other clients). How about:
```suggestion
   * Updates the current client (i.e. browser tab) to the latest version that is ready for activation.
```